### PR TITLE
implemet group query and remove group members api

### PR DESCRIPTION
**Pull request by keylorsdu**
implemet group query and remove group members api. and write some tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ Releases are available with Maven:
 <dependency>
     <groupId>com.urswolfer.gerrit.client.rest</groupId>
     <artifactId>gerrit-rest-java-client</artifactId>
-    <version>0.8.13</version>
+    <version>0.8.14</version>
```
Comment (uwolfer) [inline README.md]: Can you please revert version changes in your PR?

File: appveyor.yml [hunk 2]
```diff
@@ -1,12 +1,11 @@
 environment:
-  MVN_VERSION: 3.5.0
+  JAVA_HOME: C:\Program Files\Java\jdk1.8.0
```
Comment (uwolfer) [inline appveyor.yml]: Is this (and following) change required?

File: src/main/java/com/urswolfer/gerrit/client/rest/http/accounts/AccountsRestClient.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public SuggestAccountsRequest suggestAccounts(String query) throws RestApiExcept
     }
 
     private List<AccountInfo> getSuggestAccounts(String queryPart) throws RestApiException {
-        String request = String.format("/accounts/?%s", queryPart);
+        String request = String.format("/accounts/?suggest&%s", queryPart);
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/accounts/AccountsRestClient.java]: Is this change required?

File: src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupApiRestClient.java [hunk 4]
```diff
@@ -170,4 +170,13 @@ public void removeGroups(String... groups) throws RestApiException {
 
         gerritRestClient.postRequest(restPath, json);
     }
+    
+    @Override
+    public void removeMembers(String... members) throws RestApiException {
+    	String restPath = getRequestPath() + "/members.delete";
+        Map<String, List<String>> memberMap =
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupApiRestClient.java]: Can you please check formatting of the whole change (use 4 spaces for 1 tab).

File: src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClient.java [hunk 5]
```diff
@@ -133,4 +133,50 @@ public ListRequest list() {
...
+			public List<GroupInfo> get() throws RestApiException {
+				List<GroupInfo> list = GroupsRestClient.this.query(this);
+				return list;
+			}
+		};
+    }
+    
+    protected List<GroupInfo> query(QueryRequest queryRequest) throws RestApiException {
+    	 String query = "";
+		 if(!Strings.isNullOrEmpty(queryRequest.getQuery())) {
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClient.java]: Please add a space after if.

File: src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClient.java [hunk 6]
```diff
@@ -133,4 +133,50 @@ public ListRequest list() {
...
+         if (queryRequest.getLimit() > 0) {
+             query = UrlUtils.appendToUrlQuery(query, "limit=" + queryRequest.getLimit());
+         }
+         if (queryRequest.getStart() > 0) {
+             query = UrlUtils.appendToUrlQuery(query, "start=" + queryRequest.getStart());
+         }
+         if (!queryRequest.getOptions().isEmpty()) {
+             throw new NotImplementedException();
+         }
+
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClient.java]: Remove one empty line here.

File: src/test/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClientTest.java [hunk 7]
```diff
@@ -231,4 +233,138 @@ public TestListRequest withStart(int start) {
...
+    public void testQueryGroupsWithParameter(GroupQueryTestCase testCase) throws Exception {
+        testCase.execute().verify();
+    }
+
+    @DataProvider(name = "QueryGroupsTestCases")
+    public Iterator<GroupQueryTestCase[]> queryGroupTestCases() throws Exception {
+    	 return Iterables.transform(Arrays.asList(
+    	            queryTestCase().withQueryParameter(
+    	                    new TestQueryRequest()
+    	                    ).expectUrl("/groups/"),
```
Comment (uwolfer) [inline src/test/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClientTest.java]: Please check formatting here (indent).

File: src/test/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClientTest.java [hunk 8]
```diff
@@ -231,4 +233,138 @@ public TestListRequest withStart(int start) {
...
+            return expectedUrl;
+        }
+    }
+    private final static class TestQueryRequest {
+        private EnumSet<ListGroupsOption> options =
+            EnumSet.noneOf(ListGroupsOption.class);
+        private List<String> projects = new ArrayList<String>();
+        private List<String> groups = new ArrayList<String>();
+
+
```
Comment (uwolfer) [inline src/test/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClientTest.java]: Remove this empty line (and some of the following where there is more than one).

File: src/main/java/com/urswolfer/gerrit/client/rest/http/accounts/AccountsRestClient.java (hunk 3 above)
Comment (keylorsdu) [reply src/main/java/com/urswolfer/gerrit/client/rest/http/accounts/AccountsRestClient.java]: with param "suggest", detail information like name etc. will be included in the query result. or only account id is returned.

**Comment by uwolfer** [comment]
I have merged your change: https://github.com/uwolfer/gerrit-rest-java-client/commit/2a15a7fde4531f9c2f80b687b13aec6cde41a0ba

I have changed tabs to spaces and added one more test case. If you depend on a new release, you could use https://jitpack.io/ or tell me so I could release it.

**Comment by keylorsdu** [comment]
thanks for your sincere help.
